api/web3server_utils.go

```diff
@@ -306,19 +305,65 @@ func (svr *Web3Server) parseBlockRange(fromStr string, toStr string) (from uint6
 	return
 }
 
-func (svr *Web3Server) isContractAddr(addr string) (bool, error) {
+func (svr *Web3Server) actionTypeByAddr(addr string) (actionType, error) {
+	if addr == address.StakingProtocolAddr {
+		return _stakingAction, nil
+	}
 	if addr == "" {
-		return true, nil
+		return _executionAction, nil
 	}
 	ioAddr, err := address.FromString(addr)
 	if err != nil {
-		return false, err
+		return "", err
 	}
 	accountMeta, _, err := svr.coreService.Account(ioAddr)
-	if err != nil {
-		return false, err
+	if err != nil && accountMeta.IsContract {
+		return _executionAction, nil
+	}
+	return _transferAction, nil
+}
+
+func loadStakingAction(data []byte, core *iotextypes.ActionCore) error {
```

Comment:
IMO action package is a better place for this code
web3 format transfer/execution are handled in there, web3 staking are also handled in the new `action.ToRLP()`, kinda weird to leave this part specifically in web3 package